proms-resources/frontend/src/main/frontend/src/proms/WelcomeMessageConfiguration.jsx

```diff
@@ -0,0 +1,177 @@
+
+  // Fetch saved admin config settings
+  let getWelcomeMessage = () => {
+    fetch('/Proms/WelcomeMessage.json')
+      .then((response) => response.ok ? response.json() : Promise.reject(response))
+      .then((json) => {
+        setFetched(true);
+        setPath(json["@path"]);
+        setWelcomeMessage(json.text || "");
+      })
+      .catch(setError);
+  }
+
+  // Submit function
+  let handleSubmit = (event) => {
+
+    // This stops the normal browser form submission
+    event && event.preventDefault();
+
+    // Build formData object.
+    // We need to do this because sling does not accept JSON, need url encoded data
+    let formData = new URLSearchParams();
+    formData.append('text', welcomeMessage);
+
+    // Use native fetch, sort like the XMLHttpRequest so no need for other libraries.
+    fetch("/Proms/WelcomeMessage",
+      {
+        method: 'POST',
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/x-www-form-urlencoded'
+        },
+        body: formData
+      })
+      .then((response) => {
+
+        // Important note about native fetch, it does not reject failed
+        // HTTP codes, it'll only fail when network error
+        // Therefore, you must handle the error code yourself.
+        if (!response.ok) {
+          throw Error(response.statusText);
+        }
+
+        setIsSaved(true);
+      })
+      .catch(setError);
+  }
+
+  if (!fetched) {
+    getWelcomeMessage();
+  }
+
+  return (
+    <Card>
+      <CardHeader
+        title="Welcome Message text"
+        titleTypographyProps={{variant: "h6"}}
+      />
+      <CardContent>
+        {error && <Typography color='error'>{error}</Typography>}
+        <Typography>
+          In the welcome message, use APP_NAME to refer to the name configured for the application.
+          On the Patient identification screen, all occurrences of APP_NAME will appear as {appName}.
+        </Typography>
+        <form onSubmit={handleSubmit}>
+          <List>
+            <ListItem key="form">
+              { welcomeMessage != undefined &&
+                <Grid
+                  container
+                  spacing={2}
+                  direction={breakpoint ? "column" : "row"}
+                  justifyContent="center"
+                  alignItems="flex-start"
+                >
+                  <Grid item xs={12} sm={8} md={6}>
+                    <MarkdownText value={welcomeMessage} height={350} preview="edit" onChange={value => { event.preventDefault(); setWelcomeMessage(value); }} />
+                  </Grid>
+                  <Grid item xs={12} sm={8} md={6}>
+                    <Card>
+                      <CardHeader
+                        className={classes.previewHeader}
+                        title="Preview"
+                        titleTypographyProps={{variant: "overline"}}
+                      />
+                      <CardContent>
+                        <FormattedText>
```

Comment:
The preview font appears too large compared to the editor font, let's use a smaller one.
```suggestion
                        <FormattedText variant="body2">
```

Reply:
Done in bb8fe07.